File: abc/abc124/abc124_c.cs
// https://atcoder.jp/contests/abc124/submissions/8434239

using System;

namespace CSharp
{
    class Program
    {
        public static void Main()
        {
            var S = Console.ReadLine();

            var result1 = 0;
            var result2 = 0;
            for (int i = 0; i < S.Length; i++)
            {
                if ((i % 2 == 0 ? '0' : '1') != S[i])
                {
                    result1++;
                }
                if (((i + 1) % 2 == 0 ? '0' : '1') != S[i])
                {
                    result2++;
                }
            }
            Console.WriteLine(Math.Min(result1, result2));
        }
    }
}

File: abc/abc124/abc124_c.fs
// https://atcoder.jp/contests/abc124/submissions/12208668

let S = stdin.ReadLine()

let solver =
    let rec loop r1 r2 i =
        if i = S.Length then min r1 r2
        elif i % 2 = 0 then
            loop
                (if S.[i] = '0' then r1
                 else r1 + 1)
                (if S.[i] = '1' then r2
                 else r2 + 1) (i + 1)
        else
            loop
                (if S.[i] = '1' then r1
                 else r1 + 1)
                (if S.[i] = '0' then r2
                 else r2 + 1) (i + 1)
    loop 0 0 0

solver |> stdout.WriteLine
